### PR TITLE
Attribute ScoringProfile.FunctionAggregation as nullable

### LINKED ISSUE #16570: [BUG] [Azure.Search.Documents] Scoring profiles with functions creates exception
**Describe the bug**
I want to create s scoring profile without any functions on an index.

**Expected behavior**
This should work.

**Actual behavior (include Exception or Stack Trace)**
An **ArgumentOutOfRangeException** is thrown with message:

    Unknown ScoringFunctionAggregation value. 

**To Reproduce**
The following code shows the problem:

	public class Document
	{
	    [SimpleField(IsKey = true)]
	    public string ID { get; set; }
	
	    [SearchableField(IsSortable = false, IsFilterable = true)]
	    public string Title { get; set; } = string.Empty;
	}
	
	static async Task Main(string[] args)
	{
	    const string indexname = "testindex";
	    const string profilename = "testprofile";
	
	    var IndexClient = 
	        new SearchIndexClient
	            (
	                new Uri("https://myurl"), 
	                new AzureKeyCredential("mykey")
	            );
	
	
	    FieldBuilder bulder = new FieldBuilder();
	    SearchIndex index = new SearchIndex(indexname, bulder.Build(typeof(Document)));
	
	    index.DefaultScoringProfile = profilename;
	
	    index.ScoringProfiles.Add(new ScoringProfile(profilename)
	    {
	        TextWeights = new TextWeights(new Dictionary<string, double>()
	        {
	            { nameof(Document.Title), 2 },
	        }),
	        // Setting FunctionAggregation seems to solve the problem
	        // but I do not understand why. I do not see any difference in
	        // Azure with this, but the exception goes away.
	        // FunctionAggregation = ScoringFunctionAggregation.Sum
	    });
	
	    // Next line will throw an System.ArgumentOutOfRangeException with the message:
	    // "Unknown ScoringFunctionAggregation value."
	    await IndexClient.CreateIndexAsync(index);
	
	}

**Environment:**
 - Azure.Search.Documents 11.1.1
 - Windows 10, .NET Core 3.1.9
 - Visual Studio 16.

### PR DESCRIPTION
**Pull request by heaths**
May resolve #16570, but the issue described didn't repro. That said, based on the response from the service, the field should've been attributed as nullable.

### CODE REVIEW

**Comment by heaths** [comment]
@tg-msft I'm torn whether we should take this or not at this point with the impending release. I wasn't able to repro the issue with the latest source (this may have been caused by an older autorest.csharp external package, since our source hadn't changed in any significant way), but the swagger `ScoringProfile.functionAggregation` should've been marked `x-nullable`. So while I couldn't repro the bug, this is fixing the swagger. All that happened with 11.2.0-beta.1-era source was that in debug builds, an exception was thrown saying a non-nullable field was null. That gets compiled out in release builds.

File: sdk/search/Azure.Search.Documents/tests/SearchIndexClientTests.cs [hunk 1]
```diff
@@ -18,7 +18,7 @@ namespace Azure.Search.Documents.Tests
     public class SearchIndexClientTests : SearchTestBase
     {
         public SearchIndexClientTests(bool async, SearchClientOptions.ServiceVersion serviceVersion)
-            : base(async, serviceVersion, null /* RecordedTestMode.Record /* to re-record */)
+            : base(async, serviceVersion, RecordedTestMode.Record /* to re-record */)
```
Comment (bleroy) [inline sdk/search/Azure.Search.Documents/tests/SearchIndexClientTests.cs:21]: Did you forget to revert that or is that deliberate?

**Comment by bleroy** [comment]
To clarify, is leaving it as-is breaking?

File: sdk/search/Azure.Search.Documents/tests/SearchIndexClientTests.cs (hunk 1 above)
Comment (heaths) [reply sdk/search/Azure.Search.Documents/tests/SearchIndexClientTests.cs:21]: Nope. Good catch (though it seems CI caught it pretty quickly, too).

**Comment by heaths** [comment]
With 11.2.0-beta.1, this shouldn't even happen (the issue reported, that is). There's a DEBUG-conditional method that will throw if `null` is received, but it's not present in release builds. The `ArgumentOutOfRangeException` would only occur if the service returns a value that is not supported by this *real* enum (i.e. not extensible). And from the source code I replicated from the bug, that doesn't appear to be the case.

**Review by tg-msft — APPROVED** [review]
Since the next release is a preview, I don't see any harm in shipping a more correct swagger tweak even though we've fixed other bugs to prevent it from surfacing the way it used to.  I'm fine if the service folks agree this should be nullable and plan to update their swagger when they can.

**Comment by heaths** [comment]
> if the service folks agree this should be nullable

That's what the service returns now. According to @brjohnstmsft, the OData library they use returns explicit nulls by default. I think a longer-term conversation is to make sure the swagger reflects this properly, because issues like this (or at least for this related issue, what *was* likely the case) keep popping up. In a recent swagger review, I called out all the places where nulls would be a problem and it was updated before being merged.

**Pull request merged**